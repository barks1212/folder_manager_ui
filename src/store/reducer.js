import folderData from '../data/folders.json';
import * as actionTypes from './actions';

const initialState = {
    folderData,
    sort: null,
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case actionTypes.SET_SORT:
            if (action.sort === 'name') {
                return {
                    ...state,
                    sort: 'name'
                }
            }
            if (action.sort === 'date') {
                return {
                    ...state,
                    sort: 'date'
                };
            };
            break;
        case actionTypes.SORT_FILES:
            let sortedState;
            if (action.sort === 'name') {
                sortedState = folderData.sort((a, b) => {
                    if (a.name < b.name) return -1;
                    if (a.name > b.name) return 1;
                    return 0;
                });
                return {
                    ...state,
                    folderData: sortedState
                };
            };
            break;
        default:
            return state;
    };
};

export default reducer;