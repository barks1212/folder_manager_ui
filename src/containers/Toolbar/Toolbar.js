import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import Dropdown from '../../components/Dropdown/Dropdown';
import Input from '../../components/Input/Input';

import classes from './Toolbar.module.css';
import * as actionTypes from '../../store/actions';
import {noop} from '../../utils/utils';

class Toolbar extends Component {
    render() {
        const { setSort } = this.props;

        return (
            <header className={classes.Toolbar}>
                <Dropdown setSort={setSort} />
                <Input filterFiles={noop} />
            </header>
        );
    };
};


const mapDispatchToProps = dispatch => ({
    setSort: sort => dispatch({type: actionTypes.SET_SORT, sort: sort})
});

Toolbar.propTypes = {
    setSort: PropTypes.func
};

Toolbar.defaultProps = {
    setSort: noop
};

export default connect(null, mapDispatchToProps)(Toolbar);
