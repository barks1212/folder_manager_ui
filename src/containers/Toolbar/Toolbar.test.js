import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render, cleanup } from '@testing-library/react'

import Toolbar from './Toolbar';
import reducer from '../../store/reducer';

const store = createStore(reducer);

afterEach(cleanup);

describe('Toolbar', () => {
    it('should render', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <Toolbar />
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    })
});
