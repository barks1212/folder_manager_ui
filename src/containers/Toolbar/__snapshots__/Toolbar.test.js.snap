// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should render 1`] = `
<DocumentFragment>
  <header
    class="Toolbar"
  >
    <select>
      <option
        disabled=""
        selected=""
        value="Sort By"
      >
        Sort by
      </option>
      <option
        value="name"
      >
        Name
      </option>
      <option
        value="added"
      >
        Date
      </option>
    </select>
    <div
      class="InputWrapper"
    >
      <i
        class="fa fa-search"
      />
      <input
        placeholder="Filename..."
      />
    </div>
  </header>
</DocumentFragment>
`;
